iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -130,11 +130,10 @@ def addLanguage(self, langCode):
     def addELR(self, elr):
         prefix = self.roleMap.getPrefix(elr)
         if self.taxonomyData.setdefault("roleDefs",{}).get(prefix, None) is None:
-            rt = self.dts.roleTypes[elr]
-            label = elr
-            if len(rt) > 0:
-                label = rt[0].definition
-            self.taxonomyData["roleDefs"].setdefault(prefix,{})["en"] = label
+            rts = self.dts.roleTypes.get(elr, [])
+            label = next((rt.definition for rt in rts if rt.definition is not None), None)
```

Comment:
I guess all `roleType` must be S-Equal anyway, so we could just keep the original logic?